docs/Simulation/Software-RTL-Simulation.rst

```diff
@@ -0,0 +1,139 @@
+For example:
+
+.. code-block:: shell
+
+    make run-asm-tests
+    make run-bmark-tests
+
+
+.. Note:: Before running the pre-packaged suites, you must run the plain ``make`` command, since the elaboration command generates a Makefile fragment that contains the target for the pre-packaged test suites. Otherwise, you will likely encounter a Makefile target error. 
+
+
+.. _sw-sim-custom:
+Simulating A Custom Project
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you later create your own project, you can use environment variables to build an alternate configuration.
+
+In order to construct the simulator with our custom design, we run the following command within the simulator directory:
+
+.. code-block:: shell
+
+    make SBT_PROJECT=... MODEL=... VLOG_MODEL=... MODEL_PACKAGE=... CONFIG=... CONFIG_PACKAGE=... GENERATOR_PACKAGE=... TB=... TOP=...
+
+Each of these make variables correspond to a particular part of the design/codebase and are needed so that the make system can correctly build and make a RTL simulation.
+
+The ``SBT_PROJECT`` is the ``build.sbt`` project that holds all of the source files and that will be run during the RTL build.
+
+The ``MODEL`` and ``VLOG_MODEL`` are the top-level class names of the design.
+
+Normally, these are the same, but in some cases these can differ (if the Chisel class differs than what is emitted in the Verilog).
+
+The ``MODEL_PACKAGE`` is the Scala package (in the Scala code that says ``package ...``) that holds the ``MODEL`` class.
+
+The ``CONFIG`` is the name of the class used for the parameter Config while the ``CONFIG_PACKAGE`` is the Scala package it resides in.
+
+The ``GENERATOR_PACKAGE`` is the Scala package that holds the Generator class that elaborates the design.
+
+The ``TB`` is the name of the Verilog wrapper that connects the ``TestHarness`` to VCS/Verilator for simulation.
+
+Finally, the ``TOP`` variable is used to distinguish between the top-level of the design and the ``TestHarness`` in our system.
+For example, in the normal case, the ``MODEL`` variable specifies the ``TestHarness`` as the top-level of the design.
+However, the true top-level design, the SoC being simulated, is pointed to by the ``TOP`` variable.
+This separation allows the infrastructure to separate files based on the harness or the SoC top level.
+
+Common configurations of all these variables are packaged using a ``SUB_PROJECT`` make variable.
+Therefore, in order to simulate a simple Rocket-based example system we can use:
+
+
+.. code-block:: shell
+
+    make SUB_PROJECT=yourproject
+    ./simulator-<yourproject>-<yourconfig> ...
+
+
+All `Make` targets that can be applied to the default example, can also be applied to custom project using the custom environment variables. For example, the following code example will run the RISC-V assembly benchmark suite on the BOOM subproject:
```

Comment:
You need to use a different example here since the boom configs moved up into Chipyard. A better example would be `hwacha`